src/main.rs

```diff
@@ -645,7 +645,7 @@ fn build_basin_reconfig(
     timestamping_uncapped: Option<&bool>,
     create_stream_on_append: Option<&bool>,
     create_stream_on_read: Option<&bool>,
-) -> (Option<StreamConfig>, Vec<String>) {
+) -> (StreamConfig, Vec<String>) {
```

Comment:
Why switch away from `Option`?

Reply:
thats only main change here! if its an Option, you can specify them all just fine, but it doesnt work as the result is always None as per my testing so far

Reply:
Ah all the fields on `StreamConfig` are `Option`al